physical/raft/fsm.go

```diff
@@ -606,86 +609,36 @@ func (f *FSM) Restore(r io.ReadCloser) error {
 		return nil
 	}
 
-	snapMeta := r.(*boltSnapshotMetadataReader).Metadata()
-
-	protoReader := NewDelimitedReader(r, math.MaxInt32)
-	defer protoReader.Close()
+	snapshotInstaller, ok := r.(*boltSnapshotInstaller)
+	if !ok {
+		return errors.New("expected snapshot installer object")
+	}
 
 	f.l.Lock()
 	defer f.l.Unlock()
 
-	// Delete the existing data bucket and create a new one.
-	f.logger.Debug("snapshot restore: deleting bucket")
-	err := f.db.Update(func(tx *bolt.Tx) error {
-		err := tx.DeleteBucket(dataBucketName)
-		if err != nil {
-			return err
-		}
-
-		_, err = tx.CreateBucket(dataBucketName)
-		if err != nil {
-			return err
-		}
-
-		return nil
-	})
-	if err != nil {
-		f.logger.Error("could not restore snapshot: could not clear existing bucket", "error", err)
+	// Close the db file
```

Comment:
This prevents any writes from happening while doing a restore, right? Maybe we should add that to the comment.

Reply:
Added to the function documentation.